SystemManager/Src/SM.cpp

```diff
@@ -122,14 +123,58 @@ void SystemManager::AMOperationTask(void* pvParameters)
 //     }
 // }
 
-// void SystemManager::PMOperationTask(void *pvParameters)
-// {
-//     TickType_t xNextWakeTime;
-//     xNextWakeTime = xTaskGetTickCount();
-//     while (true) {
-//         PM_instance.execute();
-//         vTaskDelayUntil(&xNextWakeTime, SM::PM_PERIOD_MS);
-//     }
-// }
+void SystemManager::PMOperationTask(void *pvParameters)
+{
+    PM::PathManager* path_manager = (PM::PathManager*)pvParameters;
+
+    const PM::SM_PM_Commands* pm_instructions;
+    osStatus sm_pm_queue_status; 
+
+    TickType_t xNextWakeTime;
+    xNextWakeTime = xTaskGetTickCount();
+    while (true) {
+         // Read MessageQ from SM
+        uint8_t* msg_priority = 0;
+        // Arbitrary 3ms timeout as no message is no problem, gives AM time to run
+        sm_pm_queue_status = osMessageQueueGet(path_manager->getSmPmQueue(), &pm_instructions, msg_priority, 3);
+
+        // if (sm_pm_queue_status != osOK){
+        //     // do something 
+        // }
+
+
+        // Clear MessageQ from SM
+        osMessageQueueReset(path_manager->getSmPmQueue());
+
+        //if (path_manager->getUsePmFlag().readFlag()) {
+
+        path_manager->setSmStruct(*pm_instructions); 
+
```

Comment:
Can you use the sm_pm_queue status to check if there is new data from SM and only update this if there is new data? I worry that PM takes multiple execute calls to get through a full control loop.